financial-services-accelerator/components/org.wso2.financial.services.accelerator.common/src/main/java/org/wso2/financial/services/accelerator/common/policy/utils/FilterPolicyUtils.java

```diff
@@ -0,0 +1,161 @@
+import org.apache.http.entity.StringEntity;
+import org.json.JSONObject;
+import org.wso2.financial.services.accelerator.common.constant.FinancialServicesConstants;
+import org.wso2.financial.services.accelerator.common.exception.FinancialServicesException;
+import org.wso2.financial.services.accelerator.common.exception.FinancialServicesRuntimeException;
+import org.wso2.financial.services.accelerator.common.util.HTTPClientUtils;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.util.Base64;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+
+/**
+ * Filter Policy Utils.
+ */
+public class FilterPolicyUtils {
+
+    private static final Log log = LogFactory.getLog(FilterPolicyUtils.class);
+
+    /**
+     * Extract string payload from request object.
+     *
+     * @param request The request object
+     * @return String payload
+     * @throws FinancialServicesException Payload read errors
+     */
+    public static String getStringPayload(HttpServletRequest request) throws FinancialServicesException {
+        try {
+            return IOUtils.toString(request.getInputStream());
+        } catch (IOException e) {
+            log.error("Error while extracting the payload", e);
+            throw new FinancialServicesException("Error while extracting the payload", e);
+        }
+    }
+
+    /**
+     * Method to invoke external service call with basic auth.
+     * @param propertyMap
+     * @param payload
+     * @return
+     */
+    public static JSONObject invokeExternalServiceCall(Map<String, Object> propertyMap, String payload) {
+        try {
+            HttpPost httpPost = new HttpPost(propertyMap.get("endpoint").toString());
+            StringEntity params = new StringEntity(payload);
+            httpPost.setEntity(params);
+            httpPost.setHeader(FinancialServicesConstants.CONTENT_TYPE_TAG,
+                    FinancialServicesConstants.JSON_CONTENT_TYPE);
+            httpPost.setHeader(FinancialServicesConstants.ACCEPT, FinancialServicesConstants.JSON_CONTENT_TYPE);
+
+            Map<String, String> securityData = (Map<String, String>) propertyMap.get("security");
+            if ("basic-auth".equals(securityData.get("type"))) {
+                setBasicAuthHeader(httpPost, securityData.get("username"), securityData.get("password"));
+            } else if ("oauth2".equals(securityData.get("type"))) {
+                setOauth2AuthHeader(httpPost, securityData.get("tokenEP"), securityData.get("clientId"),
+                        securityData.get("clientSecret"));
+            }
+
+            CloseableHttpResponse response = HTTPClientUtils.getHttpClient().execute(httpPost);
+
+            if (response.getStatusLine().getStatusCode() != 200) {
+                throw new FinancialServicesException("Error occurred while invoking the external service");
```

Comment:
Shall we log the status code and the error response here

Reply:
Fixed in https://github.com/wso2/financial-services-accelerator/pull/345/commits/edd4f98b0226cede51cf80f6ae7ac8549fdac9b7